runtime/src/accounts.rs

```diff
@@ -246,13 +246,21 @@ impl AccountsDB {
         )))
     }
 
-    pub fn get_vote_accounts(&self, fork: Fork) -> Vec<Account> {
+    fn get_vote_accounts(&self, fork: Fork) -> Vec<Account> {
         self.index_info
             .vote_index
             .read()
             .unwrap()
             .iter()
-            .filter_map(|pubkey| self.load(fork, pubkey, true))
+            .filter_map(|pubkey| {
+                let account = self.load(fork, pubkey, true);
+                if let Some(account) = account {
+                    if account.tokens != 0 {
+                        return Some(account);
```

Comment:
@rob-solana maybe this will better describe, so if I update the vote index to contain accounts on every fork as it should, then I need to filter here to remove accounts which are shielding their parents accounts where tokens == 0 otherwise I will hit the assert in the test on line 1603. I can change that to len==7, and then add another test at the accounts level to test the filtering.

Reply:
yep, that works

the important part is just ensuring that parents are unaffected by changes in children, especially after squash()

Reply:
doing the shielding at the Accounts level is just fine